crates/ruff/src/rules/flake8_comprehensions/rules/unnecessary_map.rs

```diff
@@ -177,17 +197,27 @@ pub(crate) fn unnecessary_map(
                 return;
             }
 
-            if parameters
-                .as_ref()
-                .is_some_and(|parameters| late_binding(parameters, body))
-            {
-                return;
+            if let Some(parameters) = parameters.as_ref() {
+                if late_binding(parameters, body) {
+                    return;
+                }
+
+                parameters
+                    .posonlyargs
+                    .iter()
+                    .chain(&parameters.args)
+                    .chain(&parameters.kwonlyargs)
+                    .any(|param| param.default.is_some())
+                    || parameters.vararg.is_some()
+                    || parameters.kwarg.is_some()
+            } else {
+                false
             }
         }
-    }
+    };
 
     let mut diagnostic = Diagnostic::new(UnnecessaryMap { object_type }, expr.range());
-    if checker.patch(diagnostic.kind.rule()) {
+    if checker.patch(diagnostic.kind.rule()) && !has_default_parameters {
```

Comment:
I think I'd vote not to flag this at all -- wdyt? Also should simplify the implementation.

Reply:
(Sorry for not mentioning this on the first review.)

Reply:
No problem at all :)